includes/WcfceMailProcessor.inc

```diff
@@ -189,10 +193,201 @@ class WcfceMailProcessor {
+    // attachments this boundary is not final now, hence we need to remove
+    // '--' at the end of the text (it would be added at the end of the files).
+    $eml = rtrim(preg_replace('!--' . $boundary . '--!', '--' . $boundary, $eml))
+      . $crlf
+      . rtrim($attachments) . '--';
 
     return $eml;
   }
 
+  /**
+   * Attaches files attached to a webform submission to activity.
+   *
+   * @param int $activity_id
+   *   Activity id.
+   *
+   * @return array|false
+   *   Entity data of the last file added, empty array or FALSE
+   *   in case of error.
+   *
+   * @throws \CiviCRM_API3_Exception
+   *   An exception generated by civicrm_api3().
+   */
+  public function attachWebformFilesToActivity($activity_id) {
+    if (!$activity_id) {
+      return FALSE;
+    }
+
+    $file_ids = $this->getWebformFileIds($this->message);
+
+    if (empty($file_ids)) {
+      return FALSE;
+    }
+
+    foreach ($file_ids as $file_id) {
+      $file = $this->getWebformFile($file_id);
+      $result = civicrm_api3('Attachment', 'create', [
+        'sequential' => 1,
+        'entity_table' => 'civicrm_activity',
+        'entity_id' => $activity_id,
+        'mime_type' => $file->filemime,
+        'name' => $file->filename,
+        'content' => $this->getWebformFile($file_id, 'content'),
+      ]);
+    }
+
+    return $result['values'][0] ?? [];
+  }
+
+  /**
+   * Encodes attached file contents to append them to email.
+   *
+   * @param string $headers
+   *   Email headers (used to get email boundary).
+   * @param array|null $message
+   *   (Optional) Email data. If set to NULL then $this->message would be used.
+   * @param string $crlf
+   *   (Optional) End of line character.
+   *
+   * @return string
+   *   Attached file contents (encoded), or empty string if no files found.
+   */
+  public function getWebformFilesAsEml($headers, $message = NULL, $crlf = MAIL_LINE_ENDINGS) {
+    $message = $message ?? $this->message;
+    $res = '';
+
+    // Check if files attached to a webform submission.
+    $file_ids = $this->getWebformFileIds($message);
+
+    if (empty($file_ids)) {
+      return $res;
+    }
+
+    $boundary = $crlf . '--' . $this->getEmailBoundary($headers) . $crlf;
+
+    // Attach files to email.
+    foreach ($file_ids as $file_id) {
+      $res .= $this->getWebformFile($file_id, 'content_eml', $crlf);
+      $res .= $boundary;
+    }
+
+    return $res;
+  }
+
+  /**
+   * Get ids of files attached to submitted webform.
+   *
+   * @param array|null $message
+   *   (Optional) Email data. If set to NULL then $this->message would be used.
+   *
+   * @return array
+   *   Ids of files attached to a webform submission, or empty array if no files
+   *   were attached.
+   *   The result array looks like: array(webform_field_num => file_id, ...).
+   */
+  public function getWebformFileIds($message = NULL) {
+    $message = $message ?? $this->message;
+    $fields = &$message['params']['node']->webform['components'];
```

Comment:
I don't think we need the ampersand `&` since we are not trying to modify the variable.
Same as below.

Reply:
To be honest we don't need it and I don't think this improves performance in any way.
For this example: 
 This `$message['params']['node']` is actually an object and in `PHP` this object variable holds a copy of the identifier, which points to the same object. Ref: https://www.php.net/manual/en/language.oop5.references.php so the ampersand is not needed here.

Reply:
okay will change it in this case.